acceptance/analyzer_test.go

```diff
@@ -145,34 +119,65 @@ func setupAnalyzeFixtures(t *testing.T) analyzeFixtures {
 
 	// Registry
 
+	// setup read and write images
 	someAppName := "some-app-image-" + h.RandString(10)
-	fixtures.authRegAppImage, fixtures.authRegAuthConfig = buildAuthRegistryImage(
+	fixtures.readWriteAppImage, fixtures.regAuthConfig = buildAuthRegistryImage(
 		t,
 		someAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
-	fixtures.readOnlyRegAppImage = readOnlyRegistry.RepoName(someAppName)
+	testRegistry.SetReadWrite(someAppName)
 
 	someOtherAppName := "some-other-app-image-" + h.RandString(10)
-	fixtures.authRegOtherAppImage, _ = buildAuthRegistryImage(
+	fixtures.readWriteOtherAppImage, _ = buildAuthRegistryImage(
 		t,
 		someOtherAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
+	testRegistry.SetReadWrite(someOtherAppName)
+
+	// setup read only images
+	someReadOnlyAppName := "some-readonly-app-image-" + h.RandString(10)
+	fixtures.readOnlyAppImage, _ = buildAuthRegistryImage(
+		t,
+		someReadOnlyAppName,
+		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
+		"--build-arg", "fromImage="+containerBaseImage,
+		"--build-arg", "metadata="+appMeta,
+	)
+	testRegistry.SetReadOnly(someReadOnlyAppName)
 
 	someCacheName := "some-cache-image-" + h.RandString(10)
-	fixtures.authRegCacheImage, _ = buildAuthRegistryImage(
+	fixtures.readWriteCacheImage, _ = buildAuthRegistryImage(
 		t,
 		someCacheName,
 		filepath.Join("testdata", "analyzer", "cache-image"),
+		testRegistry,
+		"--build-arg", "fromImage="+containerBaseImage,
+		"--build-arg", "metadata="+cacheMeta,
+	)
+	testRegistry.SetReadWrite(someCacheName)
+
+	someReadOnlyRegCacheImage := "some-read-only-cache-image-" + h.RandString(10)
+	fixtures.readOnlyCacheImage, _ = buildAuthRegistryImage(
+		t,
+		someReadOnlyRegCacheImage,
+		filepath.Join("testdata", "analyzer", "cache-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+cacheMeta,
 	)
-	fixtures.readOnlyRegCacheImage = readOnlyRegistry.RepoName(someCacheName)
+	testRegistry.SetReadOnly(someReadOnlyRegCacheImage)
```

Comment:
```suggestion
	testRegistry.SetReadOnly(someReadOnlyCacheImage)
```

Reply:
Fixed in the following [commit](https://github.com/buildpacks/lifecycle/pull/685/commits/7ce8bb8297e15af0aab609e3df4c462172972ef9)